### PR TITLE
Phase2-hgx174 Bug fixes

### PR DESCRIPTION
**Pull request by bsunanda**
(1) Adds a tolerance for comparison in double
(2) Provides the right dimension for corner edges

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalGeometry/src/HGCalGeometry.cc [hunk 1]
```diff
@@ -305,7 +305,7 @@ HGCalGeometry::CornersVec HGCalGeometry::getCorners(const DetId& detid) const {
 						id.iCell1,id.iCell2,true,false);
     }
     float dx = m_cellVec[cellIndex].param()[FlatHexagon::k_r];
-    float dy = m_cellVec[cellIndex].param()[FlatHexagon::k_R];
+    float dy = 0.5*m_cellVec[cellIndex].param()[FlatHexagon::k_R];
```
Comment (cvuosalo) [inline Geometry/HGCalGeometry/src/HGCalGeometry.cc]: Should a constant with an explanatory name be used here instead of the raw number?

Comment (bsunanda) [reply Geometry/HGCalGeometry/src/HGCalGeometry.cc]: Is 0.5 a common constant. Should I introduce half as a constant with value 0.5?

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cvuosalo @kpedro88 Can you sign this PR?

**Comment by kpedro88** [comment]
@bsunanda are the changes in layerClusters understood?

**Comment by bsunanda** [comment]
There were bugs in the earlier code (1) corners of cells are now better done and tested by Alexandros; (2) the neighbor code was not protected about crossing radius limits; (3) the radius as a function of Z is now  correctly done (and tested). These are the 3 changes - I am not sure how any of the changes can affect layerClusters - if these functions are used in making layer clusters, the earlier results were wrong.

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
@kpedro88 @rovere @clelange do we have a conclusion about the changes?

**Comment by rovere** [comment]
Hi all,
I looked at the changes and, given their size and the relative small statistics, I'd say they are "reasonable", i.e., I think they do not signal any problem. I have to admit, though, that I still do not understand how these changes could affect layerClusters, hence all quantities derived from them, but nothing underneath (basically RecHits).

**Comment by bsunanda** [comment]
@kpedro88 Could you please clear this PR? there is something else which has cropped up and i can investigate that once it is integrated

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
+1

**Pull request merged**